github/config.go

```diff
@@ -110,6 +112,31 @@ func (c *Config) NewRESTClient(client *http.Client) (*github.Client, error) {
 	return v3client, nil
 }
 
+func (c *Config) NewOctokitClient() (*pkg.Client, error) {
+
+	uv3, err := url.Parse(c.BaseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	if uv3.String() != "https://api.github.com/" {
+		uv3.Path = uv3.Path + "api/v3/"
+	}
+
+	octokitClient, err := pkg.NewApiClient(
+		// pkg.WithUserAgent("my-user-agent"), // Should this be set to terraform-provider-github or similar? Doesn't look like the user-agent is set for the other clients
```

Comment:
Seems like the provider is not setting the `user-agent` for the other clients, but maybe it should?